### PR TITLE
Update to SDK rc.36 & Refactorization

### PR DESCRIPTION
**Pull request by polarker**

### CODE REVIEW

File: src/provider.ts [hunk 1]
```diff
@@ -0,0 +1,352 @@
...
+  public client!: SignClient
+  public session!: SessionTypes.Struct
+
+  static async init(opts: ProviderOptions): Promise<WalletConnectProvider> {
+    const provider = new WalletConnectProvider(opts)
+    await provider.initialize()
+    return provider
+  }
+
+  // Please instantiate a provide with `init(opts)` always
```
Comment (h0ngcha0) [inline src/provider.ts]: Maybe we can have this as a private constructor?

File: src/provider.ts [hunk 2]
```diff
@@ -0,0 +1,352 @@
...
+    this.chainGroup = opts.chainGroup
+    this.permittedChain = formatChain(this.networkId, this.chainGroup)
+
+    this.methods = opts.methods ?? [...RELAY_METHODS]
+    if (this.methods.includes('alph_requestNodeApi')) {
+      this.nodeProvider = NodeProvider.Remote(this.requestNodeAPI)
+    } else {
+      this.nodeProvider = undefined
+    }
+    if (this.methods.includes('alph_requestNodeApi')) {
```
Comment (h0ngcha0) [inline src/provider.ts]:
```suggestion
    if (this.methods.includes('alph_requestExplorerApi')) {
```

**Review by h0ngcha0 — APPROVED** [review]
LGTM, Great improvements :100: 
Only two minor comments.

File: src/provider.ts (hunk 1 above)
Comment (polarker) [reply src/provider.ts]: Yeah, but it's not doable in Typescript.

Comment (h0ngcha0) [reply src/provider.ts]: Like this?
https://github.com/alephium/walletconnect/commit/7a0de824b7a0d9f2316c52cf34112d58db10aafe

**Review by Lbqds — APPROVED** [review]
LGTM

File: src/provider.ts (hunk 1 above)
Comment (polarker) [reply src/provider.ts]: I gave it try, but the compiler warned about it. It seems the warn was due to other issues. Awesomeee!

**Pull request merged**